lib/jenkins.groovy

```diff
@@ -0,0 +1,40 @@
+import groovy.json.JsonSlurper
+
+template = load "lib/template.groovy"
+filesystem = load "lib/filesystem.groovy"
+
+def createPipelineJob(jobName, gitUrl, pipelinePath, shouldPoll, jenkinsUrl, jenkinsCredentials, jenkinsScmPoll){
+    def tmpFile = UUID.randomUUID().toString() + ".xml"
+    def templateFile = shouldPoll ? "lib/templates/app-polling-pipeline.xml" : "lib/templates/app-manual-pipeline.xml"
+    def jobXML = template.transform(
+        readFile(templateFile),
+        [
+            git_url: gitUrl,
+            pipeline_script_path: pipelinePath,
+            jenkins_credentials: jenkinsCredentials,
+            jenkins_scm_poll: jenkinsScmPoll
+        ]
+    )
+    writeFile(tmpFile, jobXML)
+
+    // create the jenkins job the application pipeline
+    sh("curl --header 'Content-Type: application/xml' --data-binary @${tmpFile} ${jenkinsUrl}createItem?name=${jobName}")
+
+    sh("rm ${tmpFile}")
+}
+
+def createFreestyleJobs(gitUrl, jenkinsUrl){
+
+    git(url: gitUrl)
+
+    def jenkinsJobsPath = "./jenkinsJobs"
+    def listing = filesystem.listing(jenkinsJobsPath)
+    for (String filepath: listing) {
```

Comment:
This style of for loop may cause an issue

http://jenkins-ci.361315.n4.nabble.com/workflow-plugin-Serialisation-issues-when-iterating-over-a-list-of-strings-td4738010.html

I saw somewhere that .each is fixed in latest version of plugin but I haven't tried it.

Might be better to use 

```
for(int i = 0; i < listings.size(); i++){
  def filepath = listings[i]
}
```